db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBClusterClient.java

```diff
@@ -42,11 +45,14 @@ private Tuple<Endpoint, Exception> nodeSelection() {
         }
 
         for (InetSocketAddress seed : candidates) {
+            logger.debug("Using seed node [{}] for cluster node discovery.", seed);
             try {
-                ClusterInfo.Endpoint endpoint = attemptDiscovery(seed).get(settings.getGossipTimeout(), TimeUnit.MILLISECONDS);
-
-                if (endpoint != null) {
+                Optional<ClusterInfo.Member> optionalMember = attemptDiscovery(seed).get(settings.getGossipTimeout(), TimeUnit.MILLISECONDS);
+                if (optionalMember.isPresent()) {
+                    ClusterInfo.Member member = optionalMember.get();
+                    ClusterInfo.Endpoint endpoint = member.getHttpEndpoint();
                     Endpoint result = new Endpoint(endpoint.getAddress(), endpoint.getPort());
+                    logger.debug("Selected cluster node [{}] in state [{}] for connection attempt.", result, member.getState());
```

Comment:
This log statement with node address and state is pretty important as it shows that during a releection phase a FOLLOWER might be selected if no leader is currently available.